hazelcast/src/main/java/com/hazelcast/internal/management/TimedMemberStateFactory.java

```diff
@@ -208,6 +204,20 @@ private void createMemberState(MemberStateImpl memberState,
         memberState.setClientStats(node.clientEngine.getClientStatistics());
     }
 
+    private String getLocalCPMemberUuidSafely() {
+        if (instance.getConfig().getCPSubsystemConfig().getCPMemberCount() == 0) {
+            return null;
+        }
+        try {
+            CPMember localCPMember = instance.getCPSubsystem().getLocalCPMember();
+            return localCPMember != null ? localCPMember.getUuid().toString() : null;
+        } catch (UnsupportedOperationException e) {
```

Comment:
Alternatively, you can check cluster version here. Unless cluster version is 3.12, this method returns null. Once it's 3.12+, it should not fail with `UOE`. Not sure which one is better though.

Reply:
I decided not to duplicate this check as it seems to be specific to `HazelcastInstanceImpl`. I can probably remove `if (instance.getConfig().getCPSubsystemConfig().getCPMemberCount() == 0)` (as it seems to be another internal detail of CP subsystem) and catch any runtime exception that may be thrown by `instance.getCPSubsystem().getLocalCPMember()`.

WDYT?

Reply:
`getCPMemberCount() == 0` is not internal detail, it's publicly mentioned in ref-manual too. `CPMemberCount == 0` means CP subsystem is disabled.

Reply:
> `getCPMemberCount() == 0` is not internal detail, it's publicly mentioned in ref-manual too. `CPMemberCount == 0` means CP subsystem is disabled.

You're right. But `node.getClusterService().getClusterVersion().isLessThan(Versions.V3_12)` is definitely an internal detail that I wouldn't like to duplicate in `TimedMemberStateFactory`.

Reply:
So, I propose to leave it as is.

Reply:
👍